includes/icon-functions.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * SVG icons related functions
+ *
+ * @package TenUpScaffold
+ */
+
+/**
+ * An svg template helper.
+ *
+ * Example:
+ * tenupscaffold_svg_icon(
```

Comment:
@colorful-tones the old function had a height and width, did we lose those?

The original function looked like:
```
function svg_icon( $name = '', $opts = array() ) {
	$args = wp_parse_args( $opts, array(
		'role'   => 'img',
		'height' => '25',
		'width'  => '25',
		'class'  => 'icon-' . $name,
		'label'  => $name,
	) );

	$href           = URL . '/assets/icons/_defs/svg-defs.svg#icon-' . $name;
	$args['class'] .= ' icon';
	$label          = ! empty( $opts['label'] ) ? 'aria-label="' . $opts['label'] . '"' : $args['label'];
	$label          = 'presentation' === $args['role'] ? '' : $args['label'];

	return sprintf( '<svg role="%s" height="%s" width="%s" class="%s" aria-label="%s"><use xlink:href="%s"/></svg>', esc_attr( $args['role'] ), esc_attr( $args['height'] ), esc_attr( $args['width'] ), esc_attr( $args['class'] ), esc_attr( $label ), esc_url( $href ) );
}

```

Reply:
@timwright12 I'll add height and width in. I'm not certain which "old function" you are referring to though. 🤔

Reply:
@colorful-tones let's at least clean up the function name to just `svg_icon` here if we're aren't namespacing it under utilities anymore

Reply:
@timwright12 

In this file we're intentionally _not_ using PHP Namespacing, therefore we have to use prefixed namespacing to avoid collisions. (cc @christianc1 ).

Reply:
Just wondering about the motivation behind not using a namespace for this file.

Reply:
@ryansommers I think it mostly revolves around WordPress-specific best practices. So, in this case we're referencing what TwentyNineteen is doing with SVGs and template tags in general (but again related to WordPress). The reasoning being that it would be easier to onboard a new Engineer if we're keeping in line with what TwentySomething themes are doing.

With that said, there is no Namespacing in TwentyNineteen at all. Therefore, it could be argued that since we're already Namespacing in the 10up Theme Scaffold then we've already diverged from this WordPress-specific best practice and we should go all in?

I think Namespacing could be a valid usage here, but would like more opinions on the matter. 

I believe there is a time and place to keep in line with WordPress best practices, but there is a time and place to try and push those standards forward a bit. 😄 

cc: @christianc1 for visibility and possible feedback please.

Reply:
@colorful-tones ,

The only functions that I'd argue against scoping under a namespace are the those in `template-tags.php`.  Template tags retrieve values from the database and render them to the screen, or at least that's how they're loosely defined.  Example, if we had a custom field referencing an album cover, we might create a template tag, `the_album_cover()`.  This type of function shouldn't be namespaced as it meets the above definition and would adhere to that convention.

This file doesn't contain "template-tags", so should be namespaced for sure.

Reply:
I would even argue for template tags being namespaced as well. The reason being - naming things is hard. Before namespaces were available, the best practice was to prefix functions to avoid naming conflicts with other code. Namespaces give us that protection by default now.

All that being said, we should never name things the same as core functions if they are wrapped in a namespace. I have seen this cause issues in projects because PHP will look for the called function in the current namespace and then look in the global one - which may yield unexpected results.